bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java

```diff
@@ -558,6 +556,7 @@ public void thingUpdated(final Thing thing, ThingTrackerEvent thingTrackerEvent)
             // called from the thing handler itself, therefore
             // it exists, is initializing/initialized and
             // must not be informed (in order to prevent infinite loops)
+            // we assume the handler knows what he's doing and don't check config validity
```

Comment:
It seems that this assumption is wrong as e.g. the ZWave handler updates the configuration with values outside of the valid range. @cweitkamp shall we add a validation here, too? And what shall happen if the thing handler tries to update with an invalid configuration? Log a warning and deny replacing the thing?

Reply:
Yes, I agree. Both methods in `BaseThingHandler` which allows to update the configuration of a thing (`updateThing(Thing)` and `updateConfiguration(Configuration)`) lead to an unchecked configuration. There is no validation done.

Both `updateThing()` and `updateConfiguration()` call `ThingHandlerCallback.thingUpdated()` -> `ThingRegistry.updated()` -> `ThingTracker.thingUpdated()` of `ThingManagerImpl`.

In theorie we have to to check for a changed configuration of the thing and do the same like the `ThingResource`: call `ThingRegistry.updateConfiguration()` instead of just replacing the thing here. imo the handler has to be reinitialized if configuration will change.

Or did I miss something.

Reply:
> It seems that this assumption is wrong

I think the problem with this assumption is that the binding doesn't have access to the config description, so it isn't (easily) possible for it to do this as far as I can see. The framework only passes the values to the binding - maybe it should also make the descriptions more readily available as well now that we're enforcing these checks.